### ISSUE TITLE
Outputting of single base "contigs".

### ISSUE THREAD

**Issue by andersgs** [rrwick/Unicycler#16]
Hi Ryan.

I am currently running version `Unicycler v0.3.0`, and noticed that a number (i.e., 4) contigs were outputted as single-base contigs (i.e., `A`, `T`, `C`, and `G`). Seems baffling, but I did notice the default for `--min_fasta_length` is 1. Maybe a better default option might be 200. Not entirely sure that is the best cutoff, but it seems more sensible than 1. Also, out of curiosity, how is it possible to output a single-base "contig" (not sure it should be called a contig either)?

Apologies if this has already been addressed in a newer version. 

Cheers.

A.

**Comment by rrwick** [comment]
I agree, a single base isn't really a 'contig', though any cutoff is arbitrary. I chose to leave them in so there is a one-to-one correspondence between the graph segments in the GFA file and the contigs in the FASTA file.

The reason such short contigs exist is the graph: they may be worthless on their own but are usually important junctions in the graph. SPAdes contigs never get that short because it has sequence overlap between graph segments (the overlap size is equal to the top k-mer size). Unicycler trims off the overlap, so the contigs get shorter. In fact, 0 bp contigs are possible - they provide no sequence, just graph structure. Before I added the --min_fasta_length option, those could end up in the FASTA file too. So it could be worse 😀

I'm happy to increase the default, as you suggested. But I'm inclined to keep it relatively low... maybe 100? I'll change it in the development branch and it should make it into the next big release.
